src/dmd/dmodule.d

```diff
@@ -389,12 +386,12 @@ extern (C++) class Package : ScopeDsymbol
         if (isPkgMod != PKG.unknown)
             return;
 
-        Identifiers packages;
+        Identifier[] packages;
         for (Dsymbol s = this.parent; s; s = s.parent)
-            packages.insert(0, s.ident);
+            packages = s.ident ~ packages;
```

Comment:
the Array package does a much better job with insertion than ~ does. ~ also uses the GC directly, which dmd tries to avoid.